jenkins/agent-ctl.sh

```diff
@@ -0,0 +1,305 @@
+#!/bin/bash
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Run command with no args for usage instructions.
+
+usage() {
+  echo "Usage: $(basename "${0}") [--pr] [--real] <INSTANCE> [ACTION ...]"
+  echo '  --pr: talk to the pull-request instead of e2e server'
+  echo '  --real: use real sized instances'
+  echo 'INSTANCE: the name of the instance to target'
+  echo '  pr-: create a pr-builder-sized instance'
+  echo '  light-: create an instance for light postcommit jobs (e2e)'
+  echo '  heavy-: create an instance for heavy postcommit jobs (build)'
+  echo '  bespoke-: create a random instance for testing'
+  echo 'Actions (auto by default):'
+  echo '  attach: connect the instance to the jenkins master'
+  echo '  auto: detatch delete create update reboot attach'
+  echo '  create: insert a new vm'
+  echo '  delete: delete a vm'
+  echo '  detatch: disconnect the instance from the jenkins master'
+  echo '  reboot: reboot or hard reset the VM'
+  echo '  update: configure prerequisite packages to run tests'
+  exit 1
+}
+
+
+set -o nounset
+set -o errexit
+
+DOCKER_VERSION='1.9.1-0~wheezy'
+GO_VERSION='go1.6.2.linux-amd64.tar.gz'
+TIMEZONE='America/Los_Angeles'
+
+REAL=
+PR=
+
+# Defaults
+IMAGE='debian-7-backports'  # TODO(fejta): debian8
+SCOPES='cloud-platform,compute-rw,storage-full'  # TODO(fejta): verify
+
+if [[ -z "${1:-}" ]]; then
+  usage
+fi
+
+while true; do
+  case "${1:-}" in
+    --large)
+      REAL=yes
+      shift
+      ;;
+    --pr)
+      PR=yes
+      shift
+      ;;
+    *)
+      break
+      ;;
+  esac
+done
+
+INSTANCE="${1}"
+shift
+if [[ -n "${PR}" ]]; then
+  echo 'Talking to PR jenkins'
+  MASTER='pull-jenkins-master'
+else
+  MASTER='jenkins-master'
+fi
+
+if [[ "${INSTANCE}" =~ light- ]]; then
+  KIND='light'
+elif [[ "${INSTANCE}" =~ heavy- ]]; then
+  KIND='heavy'
+elif [[ "${INSTANCE}" =~ pr- ]]; then
+  KIND='pr'
+else
```

Comment:
can we fail before we create a VM, then? I don't want to go through the process of creating and initializing a VM only to have it fail at attaching.